File: LMIS/LMIS.Web/Modules/ServerTypes/ReaderManage/BorrowColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BorrowStatusEnum } from "../Modules/ReaderManage.Borrow.BorrowStatusEnum";
import { BorrowRow } from "./BorrowRow";

export interface BorrowColumns {
    BorrowId: Column<BorrowRow>;
    BorrowNo: Column<BorrowRow>;
    UserName: Column<BorrowRow>;
    BookName: Column<BorrowRow>;
    Location: Column<BorrowRow>;
    BorrowStatus: Column<BorrowRow>;
    BorrowDate: Column<BorrowRow>;
    BorrowReturnDate: Column<BorrowRow>;
    CreateTime: Column<BorrowRow>;
    UpdateTime: Column<BorrowRow>;
}

export class BorrowColumns extends ColumnsBase<BorrowRow> {
    static readonly columnsKey = 'ReaderManage.Borrow';
    static readonly Fields = fieldsProxy<BorrowColumns>();
}

[BorrowStatusEnum]; // referenced types

File: LMIS/LMIS.Web/Modules/ReaderManage/Borrow/BorrowColumns.cs
using LMIS.Modules.ReaderManage.Borrow;
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.ReaderManage.Columns;

[ColumnsScript("ReaderManage.Borrow")]
[BasedOnRow(typeof(BorrowRow), CheckNames = true)]
public class BorrowColumns
{
    [EditLink, AlignRight]
    public long BorrowId { get; set; }
    [EditLink]
    [Width(200)]
    public string BorrowNo { get; set; }
    public string UserName { get; set; }
    public string BookName { get; set; }
    public string Location { get; set; }

    public BorrowStatusEnum BorrowStatus { get; set; }
    [Width(100)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd")]
    public DateTime BorrowDate { get; set; }
    [Width(100)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd")]
    public DateTime BorrowReturnDate { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime CreateTime { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}